### DISCUSSION TITLE
GraphQL - query only URL for media content

### DISCUSSION THREAD

**Discussion by sarathkcm** [strapi/strapi#7877] [Legacy Topics]
I discovered the awesomeness that is Graphql for strapi a few days ago, now I'm using it to build a blog.

I hit a problem when I tried to pull a list of all articles for displaying in the UI - bulk of the API response is just metadata and URLs of each image format (small,medium,large,thumbnail, original) of cover image in my BlogPost type. 

I just needed the title and a single coverImage URL  (small format)  in my response, but since the `formats` field in Media type is of type JSON, I'm unable to query only the url from the small format. As a result, the response is bloated with unnecessary data instead of having just two string fields per blog post.

Is there a reason `formats` is JSON and does not have schema?  (Could it have a proper schema, may be in future versions of strapi?)

Is there a way around for me to achieve what I'm trying to do? I'm interested in knowing how others are doing it, or if they are doing it at all.

**Comment by derrickmehaffy** [comment]
We went with the `json` field type to prevent needing extra tables in the database with relations, as you mentioned currently you cannot filter on the `json` field type and for the moment I don't have a good work around for you. :confused:

**Reply by sarathkcm** [reply]
Thank you for the response. Guess I will have to live with the extra overhead for now. Not a big issue for my use-case, just wanted confirmation that it can't be optimized further.

**Reply by derrickmehaffy** [reply]
Yup for the moment, no real good option, the payload size will be a little larger but there should be no overhead coming from the database at least (which there would be if there was some relations/table joins involved.)
